packages/next/src/server/base-server.ts

```diff
@@ -2422,15 +2421,24 @@ export default abstract class Server<ServerOptions extends Options = Options> {
       }
     }
     const { res, query } = ctx
+    const isNextPrivateUnmatchedRoute = !!query.__nextPrivateUnmatchedRoute
```

Comment:
I think we can update this to bail on rendering when `this.isRouterWorker` but still look up the not-found/404 path so that it knows if it is an app or pages path and then proxy the request to the correct worker